### PR TITLE
Refactor the build script to be more compartmentalized

### PR DESCRIPTION
**Pull request by skottler**
cc @outoftime @a-warner

### CODE REVIEW

File: build_ree.sh [hunk 1]
```diff
@@ -2,45 +2,59 @@
...
+function init() {
+  FULL_VERSION="1.8.7-2012.02"
+  FULL_NAME="ruby-enterprise-${FULL_VERSION}"
+  TEMP_DIR=$(mktemp -d)
 
-TEMP_DIR=$(mktemp -d)
+  echo "Serving from ${TEMP_DIR}"
 
-cd $TEMP_DIR
+  cd /tmp
```
Comment (outoftime) [inline build_ree.sh:12]: This doesn't seem consistent with the message above on line 10?

File: build_ree.sh [hunk 2]
```diff
@@ -2,45 +2,59 @@
...
+function download_and_patch() {
+  init
 
-curl https://rubyenterpriseedition.googlecode.com/files/${FULL_NAME}.tar.gz -s -o - | tar zxf -
+  cd $TEMP_DIR
+  curl https://rubyenterpriseedition.googlecode.com/files/${FULL_NAME}.tar.gz -s -o - | tar zxf -
+  cd $FULL_NAME/source
+  curl -o 34ba44f94a62c63ddf02a045b6f4edcd6eab4989.patch https://github.com/Genius/rubyenterpriseedition187-330/commit/34ba44f94a62c63ddf02a045b6f4edcd6eab4989.patch
+  curl -o 5384967a015be227e16af7a332a50d45e14ed0ad.patch https://github.com/Genius/rubyenterpriseedition187-330/commit/5384967a015be227e16af7a332a50d45e14ed0ad.patch
+  curl -o tcmalloc_declare_memalign_volatile.patch https://raw.githubusercontent.com/Genius/rubyenterpriseedition187-330/master/patches/tcmalloc_declare_memalign_volatile.patch
```
Comment (outoftime) [inline build_ree.sh]: I assume we can change these now that we've merged the other pull request?

File: build_ree.sh (hunk 1 above)
Comment (outoftime) [inline build_ree.sh:12]: (I think line 10 is incorrect)

**Comment by skottler** [comment]
@outoftime this should be all set now, it uses the patches in this repo and the message is updated to no longer use `$TEMPDIR`, but rather just `/tmp`.

File: build_ree.sh [hunk 3]
```diff
@@ -2,45 +2,59 @@
...
+function download_and_patch() {
+  init
 
-curl https://rubyenterpriseedition.googlecode.com/files/${FULL_NAME}.tar.gz -s -o - | tar zxf -
+  cd $TEMP_DIR
+  curl https://rubyenterpriseedition.googlecode.com/files/${FULL_NAME}.tar.gz -s -o - | tar zxf -
+  cd $FULL_NAME/source
+  curl -o 34ba44f94a62c63ddf02a045b6f4edcd6eab4989.patch https://raw.githubusercontent.com/Genius/ree-1.8.7-heroku/master/patches/34ba44f94a62c63ddf02a045b6f4edcd6eab4989.patch
+  curl -o 5384967a015be227e16af7a332a50d45e14ed0ad.patch https://raw.githubusercontent.com/Genius/ree-1.8.7-heroku/master/patches/5384967a015be227e16af7a332a50d45e14ed0ad.patch
+  curl -o tcmalloc_declare_memalign_volatile.patch https://raw.githubusercontent.com/Genius/ree-1.8.7-heroku/master/patches/tcmalloc_declare_memalign_volatile.patch
```
Comment (outoftime) [inline build_ree.sh]: Hmm, still seems weird to curl these when the files are available in the local filesystem of this repo?

**Comment by outoftime** [comment]
@skottler left you one question

**Comment by skottler** [comment]
@outoftime we should be good now

**Pull request merged**